### PR TITLE
Closes #171

### LINKED ISSUE #171: jQuery.isArray is deprecated; use Array.isArray
Quite a few `$.isArray` usages in the code that need switching to `Array.isArray`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

Not sure if the polyfill is needed, see above for browser compatibility.

### PR DESCRIPTION
**Pull request by bytestream**
Replaced deprecated `$.isArray` usage with `Array.isArray` as advised @ https://api.jquery.com/jQuery.isArray/

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray for browser compatibility.

### CODE REVIEW

File: js/core/core.internal.js [hunk 1]
```diff
@@ -271,3 +271,10 @@ var _unique = function ( src )
 	return out;
 };
 
+// Array.isArray polyfill.
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray
+if (! Array.isArray) {
+    Array.isArray = function(arg) {
```
Comment (bytestream) [inline js/core/core.internal.js:277]: Leaks into global space (if unset), is that OK? The rest of the functions in here are localised

File: old-tests/tests_onhold/4_server-side/ajax.js [hunk 2]
```diff
@@ -532,7 +532,7 @@ $(document).ready( function () {
 		},
 		function () {
 			console.log( result );
-			return $.isArray( result[0] ) && result[0].length === 35;
+			return Array.isArray( result[0] ) && result[0].length === 35;
```
Comment (bytestream) [inline old-tests/tests_onhold/4_server-side/ajax.js:535]: Not sure if polyfill from `core.internal.js` is available in test files, or if its required?

**Comment by DataTables** [comment]
Excellent - thank you for this. I think it is okay for the polyfill to drop into global space. I can't really see why that would be a problem unless someone is using that property to test for older browsers rather than using something more appropriate.

**Comment by DataTables** [comment]
I forgot to ask - are you happy with this being released under the MIT license?

**Comment by bytestream** [comment]
> 
> 
> I forgot to ask - are you happy with this being released under the MIT license?

Bit late now ;) but yes no worries haha.

**Comment by DataTables** [comment]
I could remove it if you didn't. But that you for confirming that :-)

**Pull request merged**